.eslintrc.js

```diff
@@ -21,11 +28,278 @@ module.exports = {
+				'import/no-cycle': 'off',
+
+				// enforce convention in import order
+				'import/order': [
+					'error',
+					{
+						'newlines-between': 'never',
+						groups: ['builtin', 'external', 'internal', 'parent', 'sibling', 'index'],
+						alphabetize: { order: 'asc', caseInsensitive: true },
+					},
+				],
+
+				// ensure consistent array typings
+				'@typescript-eslint/array-type': 'error',
+
+				// prefer type imports and exports
+				'@typescript-eslint/consistent-type-exports': [
+					'error',
+					{ fixMixedExportsWithInlineTypeSpecifier: true },
+				],
+				'@typescript-eslint/consistent-type-imports': ['error', { prefer: 'type-imports' }],
+
+				// enforce consistent order of class members
+				'@typescript-eslint/member-ordering': 'error',
+
+				// disallow parameter properties in favor of explicit class declarations
+				'@typescript-eslint/no-parameter-properties': 'error',
+
+				// ensure unused variables are treated as an error
+				// overrides @typescript-eslint/recommended -- '@typescript-eslint/no-unused-vars': 'warn'
+				// https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/src/configs/recommended.ts
+				'@typescript-eslint/no-unused-vars': 'error',
+			},
+		},
+		{
+			files: tsGlobs,
+			extends: ['plugin:@typescript-eslint/recommended-requiring-type-checking'],
+			rules: {
+				// ban ts-comment except with description
+				'@typescript-eslint/ban-ts-comment': [
+					'error',
+					{
+						'ts-expect-error': 'allow-with-description',
+						'ts-ignore': 'allow-with-description',
+						'ts-nocheck': true,
+						'ts-check': false,
+					},
+				],
+
+				// disable rules turned on by @typescript-eslint/recommended-requiring-type-checking which are too noisy
+				// https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/src/configs/recommended-requiring-type-checking.ts
+				'@typescript-eslint/no-unsafe-argument': 'off',
+				'@typescript-eslint/no-unsafe-assignment': 'off',
+				'@typescript-eslint/no-unsafe-call': 'off',
+				'@typescript-eslint/no-unsafe-member-access': 'off',
+				'@typescript-eslint/no-unsafe-return': 'off',
+				'@typescript-eslint/restrict-template-expressions': 'off',
+				'@typescript-eslint/unbound-method': 'off',
+
+				// force explicit member accessibility modifiers
+				'@typescript-eslint/explicit-member-accessibility': 'error',
+
+				// disallow boolean comparisons against non-boolean values
+				'@typescript-eslint/strict-boolean-expressions': ['error', { allowNullableBoolean: true }],
+			},
+		},
+
+		{
+			files: jsGlobs,
+			rules: {
+				// allow ts-nocheck in js files
+				'@typescript-eslint/ban-ts-comment': [
+					'error',
+					{
+						'ts-expect-error': 'allow-with-description',
+						'ts-ignore': 'allow-with-description',
+						'ts-nocheck': false,
+						'ts-check': false,
+					},
+				],
+			},
+		},
+
+		{
+			files: ['**/index.[jt]s?(x)', '**/*.gql.[jt]s', '**/constants.ts'],
```

Comment:
The `gql` suffix is definitely not needed as an override. I don't think the `constants` one is needed.

Reply:
removed `gql` and `constants` file matches.